src/utils.cpp

```diff
@@ -253,6 +268,38 @@ std::string utils::retrieve_url(const std::string& url,
 	return retrieve_url(url, handle, cfgcont, authinfo, body, method);
 }
 
+struct UtilsHeaderValues {
```

Comment:
I ran into some strange issues when just calling this `HeaderValues`.
It looks like it clashes with the `HeaderValues` struct in `cache.cpp` (IIRC it called `cache.cpp`'s HeaderValues destructor at the end of the `utils::retrieve_url()` scope).

Given this was an issue, it might also be an issue with the `HeaderValues` in `rss/parser.cpp`.
Not sure, I might have time again this weekend to take a bit more detailed look at it.

Reply:
I expect this was caused by violating the so-called [One Definition Rule](https://en.cppreference.com/w/cpp/language/definition#One_Definition_Rule). Fixed by putting each of the definitions in an anonymous namespace.